src/object/cli_obj.c

```diff
@@ -1422,7 +1422,7 @@ obj_iod_sgl_valid(unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
 	for (i = 0; i < nr; i++) {
 		if (iods[i].iod_name.iov_buf == NULL)
 			/* XXX checksum & eprs should not be mandatory */
-			return false;
+			return -DER_INVAL;
```

Comment:
returning a bool is definitely wrong, but i am not sure that return DER_INVAL for all cases is correct as well. @mchaarawi could you please check?

Reply:
I was the one who suggested this to be fixed this way.
this function is checking all arguments are valid, and in this case the return val is OK to be INVAL incase of an invalid argument. we might want to add more descriptive error codes in the future like INVAL_KEY/TYPE/RECX, etc.